app/extras/event_bus.rb

```diff
@@ -0,0 +1,29 @@
+class EventBus
+  include Singleton
+
+  def broadcast(event, *params)
+    listeners[event].each { |listener| listener.call(*params) }
+  end
+
+  def listen(event, &block)
+    listeners[event].add(block)
+  end
+
+  def deafen(event, &block)
+    listeners[event].delete(block)
+  end
+
+  def clear
+    @listeners = nil
+  end
+
+  # allows us to call EventBus.listen, instead of EventBus.instance.listen
+  [:broadcast, :listen, :deafen, :clear].each { |m| define_singleton_method m, ->(*args, &block) { instance.send(m, *args, &block) } }
```

Comment:
It's probably not better, per se. I have a probably unreasonable aversion to using class variables (@@something) for anything.